File: 0240-search-a-2d-matrix-ii/0240-search-a-2d-matrix-ii.java
class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
        
        int rows = matrix.length;
        int cols = matrix[0].length;
        
        int row = rows-1;
        int col = 0;
        
        while(row >= 0 && col < cols){
            
            if(matrix[row][col] == target){
                return true;
            }
            else if(matrix[row][col] < target){
                col++;
            }
            else{
                row--;
            }
        }
        return false;
        
    }
}

File: 0240-search-a-2d-matrix-ii/0240-search-a-2d-matrix-ii.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        

        rows = len(matrix)
        cols = len(matrix[0])

        row = rows-1
        col = 0

        while(row >= 0 and col < cols):

            if(matrix[row][col] < target):
                col+=1
            elif(matrix[row][col] > target):
                row-=1
            else:
                return True
        
        return False
